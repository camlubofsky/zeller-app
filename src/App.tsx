import React from 'react';
import { Box } from '@mui/material';
import { UserTypeGroup } from './components/UserTypeGroup';
import { UserGroup } from './components/UserGroup';
import { useState } from 'react';
import { useQuery } from '@apollo/client';
import { ListZellerCustomers } from './graphql/queries';

export const App: React.FC<any> = () => {
  const [selectedUserType, setSelectedUserType] = useState<string>('');

  const { loading, error, data } = useQuery(ListZellerCustomers);

  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error : {error.message}</p>;
  const users2 = data.listZellerCustomers.items;

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setSelectedUserType(event.target.value);
  };

  return (
    <Box
      sx={{
        p: '3rem',
        flexDirection: 'column',
      }}
    >
      <UserTypeGroup
        onChange={handleChange}
        selectedUserType={selectedUserType}
      />
      {selectedUserType && (
        <UserGroup
          users={users2.filter((user: any) => user.role === selectedUserType)}
          type={selectedUserType}
        />
      )}
    </Box>
  );
};
