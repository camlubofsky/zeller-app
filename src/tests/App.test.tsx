import '@testing-library/jest-dom';
import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { App } from '../App';
import { ListZellerCustomers } from '../graphql/queries';

const mocks = [
  {
    request: {
      query: ListZellerCustomers,
    },
    result: jest.fn().mockReturnValue({
      data: {
        listZellerCustomers: {
          __typename: 'ZellerCustomerConnection',
          items: [
            {
              email: 'test1',
              id: 'test1',
              name: 'Test One',
              role: 'MANAGER',
            },
            {
              email: 'test2',
              id: 'test2',
              name: 'Test Two',
              role: 'MANAGER',
            },
          ],
        },
      },
    }),
  },
];

test('Snapshot test', async () => {
  const { asFragment } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <App />
    </MockedProvider>
  );

  expect(asFragment()).toMatchSnapshot();
});
