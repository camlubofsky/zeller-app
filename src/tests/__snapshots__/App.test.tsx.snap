// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;

exports[`Snapshot test 2`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;
